src/zope/app/file/file.py

```diff
@@ -243,14 +248,20 @@ def __str__(self):
             result.append(self._data)
             next = self.next
 
-        return ''.join(result)
+        return b''.join(result)
+
+    if str is bytes:
+        __str__ = __bytes__
+    else:
+        def __str__(self):
+            return self.__bytes__().decode("iso-8859-1", errors='ignore')
```

Comment:
Right. I think it saves a little bit of brain-power when reading though, even if you know nothing about codecs.